packages/frontend/app/routes/assets.$assetId.tsx

```diff
@@ -289,12 +337,44 @@ export async function action({ request }: ActionFunctionArgs) {
 
       break
     }
+    case 'delete': {
+      const result = uuidSchema.safeParse(Object.fromEntries(formData))
+      if (!result.success) {
+        return setMessageAndRedirect({
+          session,
+          message: {
+            content: 'Invalid asset ID.',
+            type: 'error'
+          },
+          location: '.'
+        })
+      }
+
+      const response = await deleteAsset({ id: result.data.id })
+      if (!response?.success) {
+        return setMessageAndRedirect({
+          session,
+          message: {
+            content: 'Could not delete asset.',
```

Comment:
Non blocker, but maybe we can service up the error from the API here, to show why the asset could not be deleted.

Or we can add somewhere in the danger box explaining that an asset can only be deleted when not in use.